build: Get rid of outdated JS stack

vishesh opened the pull request:
- Stop using outdated JS build libraries, e.g. Traceur, Gulp
- Try to use plain NodeJS much as possible for building + testing. This will
remove the need to run Gulp + `npm install`.

stchang commented:
I'm still seeing `gulpfile.js` in `js-build`. Do I need to specify `--skip-gulp-build`?

vishesh commented:
Umm, should have been more clear. The gulpfile.js in `plain` target is empty and does nothing. https://github.com/vishesh/racketscript/blob/task/trash-old-stack/racketscript-compiler/racketscript/compiler/js-support/plain/gulpfile.js`. I should add check to skip gulp for plain target.

Gulp is still required for Babel though. Will figure that out that later. So with this PR, you can build with `plain` target and run the ES6 modules using NodeJS, instead of traceur. Same with `fixtures.rkt`.

stchang commented on tests/fixture.rkt:
```diff
@@ -74,15 +74,15 @@
+[define (run-in-nodejs fpath)
   (match-define (list in-p-out out-p-in pid in-p-err control)
-    (process* (build-path (output-directory) "node_modules" ".bin" "traceur")
+    (process* "/opt/homebrew/bin/node"
```
This was left in by accident right?

vishesh replied:
Yes. Thanks!

stchang commented on tests/fixture.rkt:
```diff
@@ -74,15 +74,15 @@
 ;; Runs module in file fpath in Racket interpreter and return
 ;; stdout and stderr produced
-(define (run-in-nodejs fpath)
+[define (run-in-nodejs fpath)
```
why is this a square bracket? :)

vishesh replied:
Woops. Probably a sign that I'm getting old for my current font size.

vishesh commented:
I've pushed changes that addresses your comments. Also noticed, that build is likely broken. I see failed tests, but overall result shows as passed (seems be going on for a while).

There is something going on with `fixtures.rkt` as well. For example, after a point all tests are crashing. But if you run them individually, they are fine.

```
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
Running in nodeJS: "/Users/vishesh/GitHub/Projects/racketscript/tests/basic/hof.rkt"
[debug] Processing library collection links at: /Users/vishesh/Library/Racket/snapshot/links.rktd
[debug] Processing library collection links at: /Applications/Racket v8.0.0.5/collects/../share/links.rktd
Module path: #<path:/Users/vishesh/GitHub/Projects/racketscript/tests/js-build/modules/hof.rkt.js>
[debug] Processing library collection links at: /Users/vishesh/Library/Racket/snapshot/links.rktd
[debug] Processing library collection links at: /Applications/Racket v8.0.0.5/collects/../share/links.rktd
racket->js failed
  context...:
   /Applications/Racket v8.0.0.5/collects/racket/system.rkt:98:0: do-process*/ports
   /Users/vishesh/GitHub/Projects/racketscript/tests/fixture.rkt:82:0: run-in-nodejs
   /Users/vishesh/GitHub/Projects/racketscript/tests/fixture.rkt:122:4
   /Applications/Racket v8.0.0.5/share/pkgs/rackunit-lib/rackunit/private/check.rkt:105:42
   /Applications/Racket v8.0.0.5/collects/racket/contract/private/arrow-higher-order.rkt:379:33
   /Users/vishesh/GitHub/Projects/racketscript/tests/fixture.rkt:215:7
   /Users/vishesh/GitHub/Projects/racketscript/tests/fixture.rkt:168:0: run-tests
   body of (submod "/Users/vishesh/GitHub/Projects/racketscript/tests/fixture.rkt" main)
[𝚌𝚛𝚊𝚜𝚑]✘
```

vishesh commented:
We were leaking port opened to run NodeJS. Fixed in commit https://github.com/vishesh/racketscript/pull/183/commits/6ca22349c5c078f421f5c25d41fe716073e4470a  of this PR.
